File: arrays-slices/1431-kids-with-greatest-number-of-candies/kids-with-candies.php
<?php


/**
 * @param Integer[] $candies
 * @param Integer $extraCandies
 * @return Boolean[]
 */
function kidsWithCandies($candies, $extraCandies)
{
    $max = max($candies);
    foreach ($candies as $i => $count) {
        $candies[$i] = $count + $extraCandies >= $max;
    }
    return $candies;
}

File: arrays-slices/1431-kids-with-greatest-number-of-candies/kids-with-candies.go
package main

func main() {

}

func kidsWithCandies(candies []int, extraCandies int) []bool {
	max := 0
	for _, v := range candies {
		if v > max {
			max = v
		}
	}

	res := make([]bool, len(candies))
	for i, v := range candies {
		if v+extraCandies >= max {
			res[i] = true
		}
	}
	return res
}
